gazelle/fileinfo.go

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2019 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gazelle
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/bazelbuild/rules_sass/gazelle/parser"
+)
+
+// FileInfo contains metadata extracted from a .sass/.scss file.
+type FileInfo struct {
+	Path, Name string
+
+	Imports []string
+}
+
+// sassFileInfo takes a dir and file name and parses the .sass file into
+// the constituent components, extracting metadata like the set of
+// imports that it has.
+func sassFileInfo(dir, name string) FileInfo {
+	info := FileInfo{
+		Path: filepath.Join(dir, name),
+		Name: name,
+	}
+
+	file, err := os.Open(filepath.Join(dir, name))
+	if err != nil {
+		log.Printf("%s: error reading sass file: %v", info.Path, err)
```

Comment:
> It's okay to return a partial FileInfo in cases where an error has occurred, but it might be good to distinguish fatal errors (file couldn't be opened) from non-fatal errors (couldn't parse).

￼Handled by returning an error in the err value in the fatal case (file not being able to be opened) and adding an "Errors" array to the FileInfo object to represent non-fatal parsing errors.